Update Docker image ref.

Sinclert opened the pull request:
This PR updates the reference to the Docker image that can be used to complete this tutorial.

The name change comes from [this refactor](https://github.com/scailfin/madminer-workflow/issues/26), and the Docker image with the updated name [is already published](https://hub.docker.com/r/madminertool/madminer-jupyter-env).

cranmer commented:
I would like to test the tutorial still runs fine after this change before merging.

Sinclert commented:
Some additional changes:

- I have replaced the path to the `software` folder (now on `/madminer`).
- I have replaced the path to the `shared` folder (the Docker WORKDIR is `/madminer` and not `/home` anymore).
- I have optimized GIT cloning by avoiding cloning the whole history.

cranmer commented:
Thank you @Sinclert 👏
